src/tcgwars/logic/impl/gen4/RisingRivals.groovy

```diff
@@ -812,16 +1059,32 @@ public enum RisingRivals implements LogicCardInfo {
           weakness M
           resistance L, MINUS20
           pokeBody "Icy Aura", {
-            text "As long as Mamoswine is your Active Pokémon, put 1 damage counter on each Active Pokémon (excluding Pokémon) between turns."
+            text "As long as Mamoswine GL is your Active Pokémon, put 1 damage counter on each Active Pokémon (excluding W Pokémon) between turns."
             delayedA {
+              before BEGIN_TURN, {
+                if(self.active && !self.owner.opposite.pbg.active.types.contains(W)||!self.types.contains(W)) {
```

Comment:
Don't check these in the if statement if you're gonna have another if statement checking them anyway. Makes it less readable for 0 gain.